### PR TITLE
Remove cppitertools + Small Cleanup

### PR DESCRIPTION
**Pull request by GNiendorf**
Small cleanup PR that removes cppitertools and unused/duplicate files. I noticed that cppitertools was only used in trkCore.cc and imported but unused in a bunch of other places and figured I would remove it and get rid of the directory.

### CODE REVIEW

**Comment by GNiendorf** [comment]
My last commit gets rid of the unused helper.cc/h and renames helper_v2.cc/h to just helper.cc/h

**Comment by GNiendorf** [comment]
Last commit makes pt cut configurable with --pt_cut flag for lst_plot_performance.py and -p flag for createPerfNumDenHists.

**Comment by GNiendorf** [comment]
@ariostas Not sure how to coordinate this, but after this PR is merged I think you'll have to update your CI a bit: Getting this error on the standalone "rm: cannot remove 'src/helper_v2.o': No such file or directory" since I renamed helper_v2 to just helper and deleted the old helper.cc that was just an outdated version of the _v2

**Comment by ariostas** [comment]
@GNiendorf Yeah, I'll change it so that it runs `make clean && echo "skip"` so that the CI doesn't stop if it doesn't delete something.

The CMSSW check still doesn't work since I haven't updated the step2 file, but hopefully in a couple of hours it should work.

**Comment by GNiendorf** [comment]
Last commit is an attempt at fixing issue #306, removing as much of the remaining legacy CPU code as I can and moving only what's used in the GNN output filling to SDLMath.h. @jkguiang Can you check the GNN ntuple to confirm there are no changes when you get a chance? Thank you.

**Comment by GNiendorf** [comment]
It looks like make_classfiles is a duplicate of this code here: https://github.com/cmstas/Software/blob/739fe8d21e76d52623f8ff70e68c2f17ae923176/makeCMS3ClassFiles/make_classfiles.py#L254

**Comment by GNiendorf** [comment]
I think this PR is ready for review when you get a chance @slava77.

**Comment by GNiendorf** [comment]
I think the variation in the CMSSW plots is because my other PR #374 was merged in recently, so this is really showing the change due to that PR.

**Comment by sgnoohc** [comment]
i don't see any problematic things. so it LGTM.

File: code/core/SDLMath.h [hunk 1]
```diff
@@ -100,5 +101,141 @@ namespace SDLMath
 
     };
 
+    inline float Phi_mpi_pi(float x)
```
Comment (slava77) [inline code/core/SDLMath.h:104]: could replace `Helix::Phi_mpi_pi` method

File: efficiency/src/helper.cc [hunk 2]
```diff
@@ -267,130 +296,34 @@ std::vector<float> getPtBounds()
     return pt_boundaries;
 }
 
-//__________________________________________________________________________________________
-/*std::vector<int> matchedSimTrkIdxs(std::vector<int> hitidxs, std::vector<int> hittypes)
+std::vector<float> getPtBounds(int mode)
```
Comment (slava77) [inline efficiency/src/helper.cc:299]: this should replace `std::vector<float> getPtBounds()` with smth like `return getPtBounds(ana.ptbound_mode)` to keep consistent ranges/definitions

**Review by slava77 — COMMENTED** [review]
there are a couple of simple updates that could be done now.
although with some good arguments could be done in a followup PR.

**Comment by GNiendorf** [comment]
Last commit fixes known issue where the y-axis would be cut off sometimes in comparison plots. Left is before, right is after my last commit.

TC_fakerate_etacoarsezoom-1

TC_fakerate_etacoarsezoom

File: efficiency/src/helper.cc (hunk 2 above)
Comment (GNiendorf) [reply efficiency/src/helper.cc:299]: This code wasn't written by me but I agree. It seems like the function without a mode input is never used so I'm just going to remove it.

**Comment by GNiendorf** [comment]
@slava77 This PR should be good to go if you don't have any comments about my last 2 commits.

**Pull request merged**